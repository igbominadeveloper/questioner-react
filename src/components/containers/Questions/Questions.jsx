import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';

import Question from '../../presentationals/Question/Question';
import {
  getQuestions,
  createNewQuestion,
} from '../../../store/modules/question';

export class Questions extends Component {
  componentDidMount() {
    this.props.getQuestions(this.props.meetup);
  }
  state = {
    question: '',
  };

  inputHandler = event => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  questionSubmitHandler = event => {
    event.preventDefault();
    this.props.createNewQuestion(
      'New Question',
      this.state.question,
      this.props.meetup,
    );
    this.setState({ question: '' });
  };

  render() {
    const { questions } = this.props;
    return (
      <Fragment>
        {questions &&
          questions.length > 0 &&
          questions.map((question, index) => (
            <Question question={question} key={index} />
          ))}
        <div className="new-question">
          <form
            action=""
            id="question-form"
            className="flex flex-column"
            onSubmit={this.questionSubmitHandler}
          >
            <div className="form-group">
              <textarea
                className="form-control"
                name="question"
                id="question"
                placeholder="Your Question"
                onChange={this.inputHandler}
                value={this.state.question}
                required
              />
            </div>
            <button className="btn-primary p-1 shadow">post</button>
          </form>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  questions: state.question.questions,
});

export default connect(
  mapStateToProps,
  { getQuestions, createNewQuestion },
)(Questions);
