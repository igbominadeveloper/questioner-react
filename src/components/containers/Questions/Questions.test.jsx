import React from 'react';
import { shallow } from 'enzyme';

import { Questions } from './Questions';
const props = {
  getQuestions: jest.fn(),
};

const component = shallow(<Questions {...props} />);

describe('Questions', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
