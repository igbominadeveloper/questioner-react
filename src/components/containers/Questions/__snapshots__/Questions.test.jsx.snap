// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions renders without errors 1`] = `
<Fragment>
  <div
    className="new-question"
  >
    <form
      className="flex flex-column"
      id="question-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="question"
          name="question"
          onChange={[Function]}
          placeholder="Your Question"
          required={true}
          value=""
        />
      </div>
      <button
        className="btn-primary p-1 shadow"
      >
        post
      </button>
    </form>
  </div>
</Fragment>
`;
